### PR TITLE
A4A: Implement reusable hosting sections (Features, Testimonials and Additional Features)

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR implements a couple of hosting section reusable components.

**Additional Features section**
Screenshot 2024-07-25 at 2 41 15 PM

**Features section**
Screenshot 2024-07-25 at 2 41 24 PM

**Testimonial section**
Screenshot 2024-07-25 at 2 41 31 PM

Depends on https://github.com/Automattic/wp-calypso/pull/92943

Closes https://github.com/Automattic/jetpack-genesis/issues/429
Closes https://github.com/Automattic/jetpack-genesis/issues/431
Closes https://github.com/Automattic/jetpack-genesis/issues/432

## Proposed Changes

* Implement the `HostingFeaturesSection` based on the `HostingSection` and `HostingOverviewFeatures` components.
* Implement the `HostingTestimonialsSection` based on the `HostingSection`.
* Implement the `HostingAdditionalFeaturesSection` based on the `HostingSection`.
* Add sample usage for easier testing.

## Testing Instructions

* Use the A4A live link and go to `/marketplace/hosting`
* Confirm that the Hosting sections are visible and correct.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/hosting-overview/hosting-v2/index.tsx [hunk 1]
```diff
@@ -14,40 +17,122 @@ export default function HostingV2() {
...
-					] }
-				/>
-			</HostingSection>
+						testimonial:
+							'This should be another WordPress.com specific testimonial. Let’s make sure it touches upon how they love the hosting, the support service, and especially the UI. This is just dummy text.',
+					},
+				] }
+			/>
 		</>
 	);
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview/hosting-v2/index.tsx]: These are sample usage and do not need to be translated.

**Review by cleacos — APPROVED** [review]
Nice!

LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16427625

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16427625).

**Pull request merged**